src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -8610,31 +8612,34 @@ private VariableComponentSyntax ParseDeconstructionComponent(bool topLevel = fal
                     designation = this.AddError(designation, ErrorCode.ERR_TypeExpected);
                 }
 
-                result = _syntaxFactory.TypedVariableComponent(type, designation);
+                result = _syntaxFactory.DeclarationExpression(type, designation);
             }
 
             return
-                topLevel ? (ComponentHasTypes(result) ? CheckFeatureAvailability(result, MessageID.IDS_FeatureTuples) : null) : result;
+                topLevel ? (TypeFoundInDeconstructionDeclarationVariables(result) ? CheckFeatureAvailability(result, MessageID.IDS_FeatureTuples) : null) : result;
         }
 
-        private static bool ComponentHasTypes(VariableComponentSyntax node)
+        /// <summary>
+        /// Checks if we can find at least one type in the deconstruction variables
+        /// </summary>
+        private static bool TypeFoundInDeconstructionDeclarationVariables(ExpressionSyntax node)
```

Comment:
This is not related to the tuples features. There are two heuristics to be confident that the user typed a deconstruction declaration: 

1. we must find the expected token (typically '=' but could be 'in' for in foreach context),
2. the variables must include at least one specified type. That's the `TypeFoundInDeconstructionDeclarationVariables` method.